docs/orchestration.asciidoc

```diff
@@ -0,0 +1,104 @@
+* When the node count of an existing NodeSet is increased, ECK increases the replicas of the corresponding StatefulSet.
+* When the node count of an existing NodeSet is decreased, ECK migrates data away from the corresponding Elasticsearch nodes to remove, then decreases the replicas of the corresponding StatefulSet, once data migration is over. Corresponding <<{p}-volume-claim-templates,PersistentVolumeClaims are automatically removed>>.
+* When an existing NodeSet is removed, ECK migrates data away from the corresponding Elasticsearch nodes to remove, decreases the StatefulSet replicas accordingly, then finally removes the corresponding StatefulSet.
+* When the specification of an existing NodeSet is updated (for example the Elasticsearch configuration, or the PodTemplate resources requirements), ECK performs a rolling upgrade of the corresponding Elasticsearch nodes. In order to do so, it follows link:https://www.elastic.co/guide/en/elasticsearch/reference/current/rolling-upgrades.html[Elasticsearch rolling upgrade best practices], to slowly upgrade Pods to the newest revision while preventing unavailability of the Elasticsearch cluster. In most cases, it corresponds to restarting Elasticsearch nodes one by one, and reusing the same PersistentVolume data. Note that some <<{p}-orchestration-limitations,cluster topologies may cause the cluster to be unavailable during the upgrade>>.
+* When an existing NodeSet is renamed, ECK performs the creation of a new NodeSet with the new name, and the removal of the old NodeSet, according to the NodeSet creation and removal patterns described above. Elasticsearch data is migrated away from the deprecated NodeSet before removal. The Elasticsearch resource <<{p}-update-strategy>> controls how many nodes can exist above or below the target node count during the upgrade.
+
+In any case, ECK ensures any StatefulSet operation is done in a way that is consistent with the Elasticsearch cluster orchestration best practices, by manipulating orchestration settings such as `discovery.seed_hosts`, `cluster.initial_master_nodes`, `discovery.zen.minimum_master_nodes`, `_cluster/voting_config_exclusions`.
+
+[id="{p}-orchestration-limitations"]
+==== Limitations
+
+Due to the way Kubernetes and StatefulSets behave, ECK orchestration has the following limitations:
+
+* Storage requirements (including volume size) of an existing NodeSet cannot be updated. link:https://github.com/kubernetes/enhancements/issues/661[StatefulSet volumes expansion is not available yet in Kubernetes]. In order to upgrade the storage size, it is however possible to create a new NodeSet, or rename the existing one. The rename will lead to the creation of a new StatefulSet with the specified storage size. The deprecated StatefulSet is removed once Elasticsearch data has been migrated away to nodes of the replacing one, automatically handled by ECK.
```

Comment:
> The deprecated StatefulSet is removed once Elasticsearch data has been migrated away to nodes of the replacing one, automatically handled by ECK.

Had to read it twice to understand. Do we need the `automatically handled by ECK` bit?

Reply:
Rephrased to 
> Elasticsearch data from the initial NodeSet will be migrated away to nodes of the replacing one